Rename calculate_on_axis_B function

akaptano opened the pull request:
As title describes, this pull request merely renames the function in permanent_magnet_helpers.py and makes some other little changes in the comments and docstring to make it clear.

andrewgiuliani commented on src/simsopt/util/permanent_magnet_helper_functions.py:
```diff
@@ -310,12 +310,14 @@ def initialize_coils(config_flag, TEST_DIR, s, out_dir=''):
+        print('Total current = ', total_current)
+        base_curves = create_equally_spaced_curves(ncoils, s.nfp, stellsym=True,
+                                                   R0=R0, R1=R1, order=order, numquadpoints=64)
+        base_currents = [(Current(total_current / ncoils * 1e-5) * 1e5) for _ in range(ncoils - 1)]
```
what's going on here?

andrewgiuliani commented on src/simsopt/util/permanent_magnet_helper_functions.py:
```diff
@@ -310,12 +310,14 @@ def initialize_coils(config_flag, TEST_DIR, s, out_dir=''):
-        total_current = Vmec(TEST_DIR / vmec_file).external_current() / (2 * s.nfp) / 8.75 / 5.69674966667
-        base_curves = create_equally_spaced_curves(ncoils, s.nfp, stellsym=True, R0=R0, R1=R1, order=order, numquadpoints=128)
-        base_currents = [(Current(total_current / ncoils * 1e-5) * 1e5) for _ in range(ncoils-1)]
+        total_current = Vmec(TEST_DIR / vmec_file).external_current() / (2 * s.nfp) / 1.311753
```
what's going on here? can we add a quick comment? + update the docstring to say what the mean modB on major radius is being set here?

akaptano replied:
Removed, and replaced by a fixed number that gets the scaling right. Updated the docstring

akaptano commented on tests/geo/test_pm_grid.py:
```diff
@@ -667,13 +667,12 @@ def test_pm_post_processing(self):
         Bfield.set_points(s_plot.gamma().reshape((-1, 3)))
-        #qfm_surf = make_qfm(s_plot, Bfield)
-        #qfm_surf = qfm_surf.surface
+        # qfm_surf = make_qfm(s_plot, Bfield)
```
I actually think this adds coverage to permanent_magnet_helpers (which is something we discussed with Bharat) so I've added the comments back in (seems to run fine and not take too long in the unit tests)

akaptano commented:
> thanks so much for splitting things up, only some really minor comments here.
> 
> It seems like you're not consistently setting up the devices with the same way, e.g.
> 
> for muse, `base_currents[0].fix_all()` is called, while for QH we scale things down and `total_current.fix_all()` is being called, which seems now like all the currents are fixed.
> 
> Can it be clarified why?

Just fixed this, now they both use the 'total_current' method. My impression has been that this is generally superior, because (I think) it is only fixing the total current through all coils (but each coil current can vary within that constraint), rather than `base_currents[0].fix_all()` explicitly fixes the current in coil 1, which might have a crappy initial value that you have now fixed forever.

landreman commented on src/simsopt/util/permanent_magnet_helper_functions.py:
```diff
@@ -332,37 +334,36 @@ def initialize_coils(config_flag, TEST_DIR, s, out_dir=''):
 
     Returns:
-        B0avg: Average magnetic field strength along 
+        B0avg (array, shape (s.nphi,)): Average magnetic field strength along 
```
Isn't B0avg a float and not an array?

landreman commented on src/simsopt/util/permanent_magnet_helper_functions.py:
```diff
@@ -332,37 +334,36 @@ def initialize_coils(config_flag, TEST_DIR, s, out_dir=''):
-        bs: MagneticField or BiotSavart class object.
-        s: plasma boundary surface.
+        bs (BiotSavart): MagneticField or BiotSavart class object.
+        s (Surface): plasma boundary surface.
```
Specifically, state that s must be a SurfaceRZFourier - otherwise the `get_rc()` method used in this function would not exist.

akaptano commented:
@landreman @andrewgiuliani I think this is ready to go but I somehow dismissed Matt's old approval of this. I will merge this in when you're ready.
